### DISCUSSION TITLE
Using only with additional_hosts

### DISCUSSION THREAD

**Discussion by AlecRust** [homeassistant-apps/app-cloudflared#190] [Q&A] [answered]
I use Home Assistant Cloud for my main Home Assistant instance, and I'm quite happy with the remote control feature it provides and supporting the project in general. I'd also like to pick up any new features added for Home Assistant Cloud users.

That said, there are other things running e.g. Home Assistant add-ons that don't support [Ingress](https://developers.home-assistant.io/docs/add-ons/presentation/#ingress), that I'd like to access remotely and cloudflared seems a great option for this.

Have you considered allowing this Cloudflared add-on to expose generic hosts but _not_ be used for the main Home Assistant remote connection i.e. without setting `external_hostname`?

I'm relatively new to exposing services externally so I'm not sure if e.g. Nginx Proxy Manager would be a better fit for my use case, but using Cloudflare certainly seems simple and secure.

**Comment by brenner-tobias** [comment]
Thanks for raising this. You are right, in the "normal" set-up, the host for home-assistant is always required, since this is the use-case, that most of the users are using. Nevertheless, there are already two options to use this add-on for ingress only without exposing home-assistant itself:

1. You can use a tunnel which is managed from the Cloudflare Dashboard (see explanation [here](https://github.com/brenner-tobias/addon-cloudflared/blob/047a0584c119293ef8de24840567d550569846ac/docs/tunnels.md)). In order to do that, you simply have to set the [tunnel_token](https://github.com/brenner-tobias/addon-cloudflared/blob/047a0584c119293ef8de24840567d550569846ac/cloudflared/DOCS.md#option-tunnel_token) in the add-on options.
2. You can use a custom config.yml with the add-on. This will override all add-on options and give you the possibility to define anything you can with Cloudflared. See explanation [here](https://github.com/brenner-tobias/addon-cloudflared/blob/047a0584c119293ef8de24840567d550569846ac/cloudflared/DOCS.md#option-custom_config-advanced-option).

Kindly let me know if this is sufficient for what you want to do.

**Comment by AlecRust** [comment]
Thanks for the reply. I assumed this was not currently possible due to this in the "Prerequisites" section of docs:

> 1. Before starting, please make sure to remove all other add-ons or configuration entries handling SSL certificates, domain names and so on (e.g. DuckDNS) and restart your Home Assistant instance.

Perhaps this should be removed to avoid potential confusion? Unless there is some specific clash with DuckDNS or Home Assistant Cloud?

Thanks for clarifying my options here. I'd rather specify my hostnames in my plugin config rather than the Cloudflare UI, but Option 1 does look the simplest to expose additional services alongside an existing Home Assistant Cloud setup.

It's good to know I can set a custom `config.yml` to achieve this but it's a shame I'd need to learn about that/set it all up manually.

I suppose I'm wondering if by default this add-on could just expose `additional_hosts` if any are set, and only expose Home Assistant if `external_hostname` is set?

This would make the add-on more generic and complimentary to Home Assistant Cloud i.e. the add-on could target:

> HA OS user wanting to securely expose services externally, including (optionally) HA itself

Instead of:

> HA OS user wanting to securely expose HA externally, including (optionally) other services

**Comment by brenner-tobias** [comment]
I get the point, thanks for the suggestion. I am working on making "external_hostname" optional in the PR #203 
Please feel free to make suggestions for documentation there. Unfortunately, I will not be able to test that properly for some time, so it might take a while to merge it in.

**Comment by brenner-tobias** [comment, marked as answer]
I just merged this into master, so it is included in the edge repository. Will be included in 3.0.2 in a couple of days.
